### PR TITLE
board: Add support for TTGO T-Display-GD32

### PR DESCRIPTION
**Pull request by tuupola**
This PR adds support for the LILYGO® TTGO T-Display-GD32  board. PR is still work in progress, documentation etc is missing.

```
$ make SOC=gd32vf103 BOARD=gd32vf103c_t_display all
```

Few  questions.

1. Is the board naming ok (`gd32vf103c_t_display`) or should it be something else?
2. Should the linker file be named `gcc_gd32vf103_flashxip.ld` like now or should it be `gcc_gd32vf103xb_flashxip.ld` like with Longan Nano? The latter version require also extra conditional sentence in `Makefile.soc.gd32vf103`.

### CODE REVIEW

**Comment by fanghuaqi** [comment]
Hi, I think it will easiler to name `gcc_gd32vf103_flashxip.ld` to not change too much in `Makefile.soc.gd32vf103`.

**Comment by tuupola** [comment]
Yes, it has a GD32VF103CBT6.

**Comment by tuupola** [comment]
Just a quick note. I have not forgotten this, just busy with other things atm.

**Comment by tuupola** [comment]
After some testing this should be ready. 

One question though. Currently there is some inconsistencies with naming of things. There is `LEDG_PIN` and `KEY_A_PIN`. There is both `LED1` and `LED_1` but only `KEY_A`. I think it would be good idea to choose one naming scheme and stick to it. 

It is still ok to break BC when bumping from `0.3.x` to `0.4.0`. I also don't think there is too much software which will break because of this, and if they do the fix is really easy.

**Comment by tuupola** [comment]
Also the board image in documentation is taken from manufacture website. If this is a problem I can take a photo of the board myself.

**Comment by fanghuaqi** [comment]
Hi @tuupola ， I prefer to user must define the following enums:

~~~c
typedef enum
{
    LED_1 = 0,
    LED_2 = 1,
    LED_3 = 2,
} led_typedef_enum;

typedef enum
{
    KEY_A = 0,
    KEY_B = 1,
    KEY_C = 2,
} key_typedef_enum;

typedef enum
{
    KEY_MODE_GPIO = 0,
    KEY_MODE_EXTI = 1
} keymode_typedef_enum;
~~~

If  you have more than 3 leds, and keys, the above should be defined.

Thanks
Huaqi

File: doc/source/design/board/gd32vf103c_t_display.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
+.. _design_board_ttgo_t_display:
+
+TTGO T-Display-GD32
+======================
```
Comment (fanghuaqi) [inline doc/source/design/board/gd32vf103c_t_display.rst:4]:
~~~
TTGO T-Display-GD32
======================
~~~
->
~~~
TTGO T-Display-GD32
===================
~~~

**Review by fanghuaqi — CHANGES_REQUESTED** [review]
Looks good to me, just a small fix for doc is required.

**Comment by tuupola** [comment]
I test compiled and uploaded everything in [application](https://github.com/Nuclei-Software/nuclei-sdk/tree/master/application) folder. There are some unrelated compiler warnings with [demo_dsp](https://github.com/Nuclei-Software/nuclei-sdk/tree/master/application/baremetal/demo_dsp) and [freertos/demo ](https://github.com/Nuclei-Software/nuclei-sdk/tree/master/application/freertos/demo). Everything works fine with my TTGO T-Display-GD32 board though. I think this should be ready for merging.

**Comment by fanghuaqi** [comment]
Done

**Pull request merged**